jerry-core/jcontext/jcontext.h

```diff
@@ -116,6 +117,84 @@ typedef struct
 } jerry_context_t;
 
 /**
+ * Description of jerry instance, which contains the meta data and buffer.
+ */
+struct jerry_instance_t
+{
+  struct
+  {
+    uint32_t heap_size; /**< size of the heap */
+    uint8_t *context_p; /**< point to the entrance of the context in buffer */
+#ifndef JERRY_SYSTEM_ALLOCATOR
+    uint8_t *heap_p; /**< point to the entrance of the heap in buffer */
+#endif /* !JERRY_SYSTEM_ALLOCATOR */
+#ifndef CONFIG_ECMA_LCACHE_DISABLE
+    uint8_t *lcache_p; /**< point to the entrance of the lcache in buffer */
+#endif /* !CONFIG_ECMA_LCACHE_DISABLE */
+  } head;
+  uint8_t buffer[]; /**< the flexible array for storing context, heap and lcache */
```

Comment:
> Hm an idea: if context_p is the first member of buffer, it could be accessed without context_p. That would be one less memory read.

Good idea indeed. It is already the first member in the buffer, plus it's size is known at compile time:

```
...
instance_p->head.context_p = instance_p->buffer;
instance_p->head.lcache_p = instance_p->head.context_p + sizeof (jerry_context_t);
...
```

So, I think we can change `jerry_instance_t` to:

```
struct jerry_instance_t
{
  struct
  {
    uint32_t heap_size; /**< size of the heap */
#ifndef JERRY_SYSTEM_ALLOCATOR
    uint8_t *heap_p; /**< point to the entrance of the heap in buffer */
#endif /* !JERRY_SYSTEM_ALLOCATOR */
#ifndef CONFIG_ECMA_LCACHE_DISABLE
    uint8_t *lcache_p; /**< point to the entrance of the lcache in buffer */
#endif /* !CONFIG_ECMA_LCACHE_DISABLE */
    jerry_context_t context;
  } head;
  uint8_t buffer[]; /**< the flexible array for storing heap and lcache */
}
```

Personally, I do like the `buffer[]` member, it's more self-documenting than without it.